source/vir/src/ast.rs

```diff
@@ -300,6 +300,14 @@ pub enum BinaryOp {
     StrGetChar,
 }
 
+/// More complex unary operations (requires Clone rather than Copy)
+/// (Below, "boxed" refers to boxing types in the SMT encoding, not the Rust Box type)
+#[derive(Clone, Debug, Serialize, Deserialize, Hash, ToDebugSNode)]
+pub enum BinaryOpr {
+    /// extensional equality ext_equal (true ==> deep extensionality)
+    ExtEq(bool, Typ),
+}
+
```

Comment:
I know this matches what we did for Unary, but is there any purpose to this split?

Reply:
It was mainly a convenience for some code that traverses AST and SST, where the code traverses into `UnaryOpr`'s op but not `Unary`'s op:

```
        ExpX::Unary(op, e1) => ok_exp(ExpX::Unary(*op, fe(env, e1)?)),
        ExpX::UnaryOpr(op, e1) => {
            let op = match op {
                UnaryOpr::Box(t) => UnaryOpr::Box(ft(env, t)?),
                UnaryOpr::Unbox(t) => UnaryOpr::Unbox(ft(env, t)?),
                ...
```

It's not a big deal, though, and we could get the same convenience in other ways if we wanted.